1) Вкладчик положил 50000 условных единиц на три разных счета в три разных банка. По первому счету выплаты составят 5% годовых, по второму – 7% годовых и по третьему 6% годовых. Известно, что через год вкладчик получил по процентам сумму в 2250 у.е. с первого и второго банка и сумму в 1400 у.е. с первого и третьего банка. Сколько условных единиц он положил на каждый счет изначально?

<div>x + y + z = 50000</div>
<div>0.05x + 0.07y = 2250</div>
<div>0.05x + 0.06z = 1400</div>

In [1]:
import numpy as np

a = np.matrix("1,1,1;0.05,0.07,0;0.05,0,0.06")
b = np.matrix("50000;2250;1400")
x = np.linalg.solve(a, b)
print(x)

[[10000.]
 [25000.]
 [15000.]]


2) На складе хранятся мобильные телефоны iPhone 6, iPhone 11, iPhone 12 моделей. Всего 1328 экземпляра. Моделей iPhone 6 на 120 меньше, чем моделей iPhone 11, и на 100 больше, чем моделей iPhone 12. Сколько моделей каждого вида на складе?

<div>x = iPhone 6</div>
<div>y = iPhone 11</div>
<div>z = iPhone 12</div>


<div>x + y + z = 1328</div>
<div>x - y = -120</div>
<div>x - z = 100</div>

In [17]:
import numpy as np

a = np.matrix("1,1,1;1,-1,0;1,0,-1")
b = np.matrix("1328;-120;100")
x = np.linalg.solve(a, b)
print(x)

[[436.]
 [556.]
 [336.]]


3) Составьте каноническое уравнение эллипсоида, проходящего через три точки A,B,C

<div>A(&#8730<span>3</span>, 0, &#8730<span>3</span>), B(&#8730<span>6, 1\2, 0</span>), c(1, 1\&#8730<span>3</span>, 1)</div>

x^2 / a^2 + y^2 / b^2 + z^2 / c^2 = 1

<div>u = 1/a^2</div>
<div>v = 1/b^2</div>
<div>w = 1/c^2</div>

<div>3u + 3w = 1</div>
<div>6u + v/4 = 1</div>
<div>u + v/3 + w = 1</div>

In [3]:
import numpy as np

a = np.matrix("3,0,3;6,0.25,0;1,0.3,1")
b = np.matrix("1;1;1")
x = np.linalg.solve(a, b)
print(1 / x)

[[13.5       ]
 [ 0.45      ]
 [ 3.85714286]]


a^2 = 13.5; b^2 = 0.45 ; c^2 = 3.8

4) Составьте уравнение параболы проходящей через заданные три точки (1,12), (3,54), (-1,2). Необходимо найти коэффициенты уравнения a,b,c


y=a⋅x^2 + bx + c

<div>a + b + c = 12</div>
<div>9a + 3b + c = 54</div>
<div>a - b + c = 1</div>

In [6]:
import numpy as np

a = np.matrix("1,1,1;9,3,1;1,-1,1")
b = np.matrix("12;54;2")
x = np.linalg.solve(a, b)
print(x)

[[4.]
 [5.]
 [3.]]


a = 4; b = 5; c = 3

5) Составьте функцию get_polynom, которая по набору точек возвращает набор коэффициентов 

In [1]:
import numpy as np


def get_polynom(coords):
    a = ''
    b = ''

    for coord in coords:
        x = coord[0]
        y = coord[1]

        a += f'{x * x},{x},1;'
        b += f'{y};'

    a = np.matrix(a.rstrip(';'))
    b = np.matrix(b.rstrip(';'))
    return np.linalg.solve(a, b)

print(get_polynom([(1, 12), (3, 54), (-1, 2)]))

[[4.]
 [5.]
 [3.]]
